security-framework/src/trust_settings.rs

```diff
@@ -0,0 +1,253 @@
+    System
+}
+
+impl Into<SecTrustSettingsDomain> for Domain {
+    fn into(self) -> SecTrustSettingsDomain {
+        match self {
+            Domain::User => kSecTrustSettingsDomainUser,
+            Domain::Admin => kSecTrustSettingsDomainAdmin,
+            Domain::System => kSecTrustSettingsDomainSystem,
+        }
+    }
+}
+
+/// Trust settings for a specific certificate in a specific domain
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum TrustSettingsForCertificate {
+    /// Not used
+    Invalid,
+
+    /// This is a root certificate and is trusted, either explicitly or
+    /// implicitly.
+    TrustRoot,
+
+    /// This is a non-root certificate but is explicitly trusted.
+    TrustAsRoot,
+
+    /// Cert is explicitly distrusted.
+    Deny,
+
+    /// Neither trusted nor distrusted.
+    Unspecified
+}
+
+impl TrustSettingsForCertificate {
+    fn new(value: i64) -> TrustSettingsForCertificate {
+        match u32::try_from(value).ok() {
+            Some(kSecTrustSettingsResultTrustRoot) => TrustSettingsForCertificate::TrustRoot,
+            Some(kSecTrustSettingsResultTrustAsRoot) => TrustSettingsForCertificate::TrustAsRoot,
+            Some(kSecTrustSettingsResultDeny) => TrustSettingsForCertificate::Deny,
+            Some(kSecTrustSettingsResultUnspecified) => TrustSettingsForCertificate::Unspecified,
+            Some(_) | None => TrustSettingsForCertificate::Invalid,
+        }
+    }
+}
+
+/// Allows access to the certificates and their trust settings in a given domain.
+pub struct TrustSettings {
+    domain: Domain,
+}
+
+impl TrustSettings {
+    /// Create a new TrustSettings for the given domain.
+    ///
+    /// You can call `iter()` to discover the certificates with settings in this domain.
+    ///
+    /// Then you can call `tls_trust_settings_for_certificate()` with a given certificate
+    /// to learn what the aggregate trust setting for that certificate within this domain.
+    pub fn new(domain: Domain) -> TrustSettings {
+        TrustSettings { domain }
+    }
+
+    /// Create an iterator over the certificates with settings in this domain.
+    /// This produces an empty iterator if there are no such certificates.
+    pub fn iter(&self) -> Result<TrustSettingsIter> {
+        let array = unsafe {
+            let mut array_ptr: CFArrayRef = ptr::null_mut();
+
+            // SecTrustSettingsCopyCertificates returns errSecNoTrustSettings
+            // if no items have trust settings in the given domain.  We map
+            // that to an empty TrustSettings iterator.
+            match SecTrustSettingsCopyCertificates(self.domain.into(), &mut array_ptr) {
+                errSecNoTrustSettings => {
+                    Ok(CFArray::from_CFTypes(&[]))
+                },
+                errSecSuccess => {
+                    Ok(CFArray::<SecCertificate>::wrap_under_create_rule(array_ptr))
+                },
+                err => Err(Error::from_code(err)),
+            }?
```

Comment:
nit: you can add `?` to `Err` arm directly, so the others won't need to be Ok-wrapped.